### PR TITLE
Add py api for sequence_slice_op

### LINKED ISSUE #13863: Sequence indexing op needed
How to do sequence indexing in paddle?
eg:
```
x is a 1-level LoDTensor:
  x.lod = [[2,    3,       2]]
  x.data = [1, 2, 3, 4, 5, 6, 7]
  x.dims = [7, 1]

id = [0, 0, 1]
output of x[id] is: [1, 3, 7]
```
In the documents, there are ops like **sequence_first_step()**, which is used to get the first step of input sequence. But this is really limited.
Thanks!

### PR DESCRIPTION
**Pull request by kuke**
Resolve #13863

### CODE REVIEW

**Review by jerrywgz — DISMISSED** [review]
LGTM

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -1901,6 +1902,72 @@ def sequence_last_step(input):
...
+    offset and subsequence length.
+
+    It only supports sequence data (LoDTensor with lod_level equal to 1).
+
+    .. code-block:: text
+    
+	- Case:
+            Given the input Variable **input**,
+	        input.data = [[a1, a2], [b1, b2], [c1, c2], [d1, d2], [e1, e2]],
+	        input.lod = [[0, 3, 5]], input.dims = (5, 2)
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py]: many public APIs, like create_lod_tensor uses length-based lod. Should the API docs be updated to length-based?

**Review by panyx0718 — DISMISSED** [review]
LG API

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (kuke) [reply python/paddle/fluid/layers/nn.py]: Agree and I made the change. Seems that now both offset-based and length-based LoD exist in the Python APIs, we'd better make them unified.

**Review by qingqing01 — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**